### PR TITLE
[New Feature] Name day plugin

### LINKED ISSUE #941: [New Feature] Name day
I would like to contribute to this project and thought that this would be a good way to start.
I would like to add a feature that can tell you when is someone's name day and if today or any other given day is someone's name day. 
I will implement it using [this](https://nameday.abalin.net/) api i found.

Since it's my first time contributing here be free to give me any advice.
I will start working on this as soon as possible.

### PR DESCRIPTION
**Pull request by AirlineDog**
This PR includes the implementation of the Name Day plugin and closes #941

The plugin gets your location to retrieve the correct data from the [API](https://nameday.abalin.net/) which has support for 20 countries.
image

You can ask for :
---

- **Today's name days**
image
- **Tomorrow's name days**
image
- **Name days for a specific date**
image
- **Name days for a specific name**
image

You can Also change your location if you wish

Some test cases are also included in the PR

**I hope you like it**

### CODE REVIEW

File: jarviscli/plugins/name_day.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+        return self.countries[self.location]
+
+    def get_location(self):
+        """
+        Get the location of the user.
+        If the user is in a supported country, the country is saved in location class variable.
+        otherwise, the user is asked to enter a country from the supported list.
+        """
+        self.jarvis.say("Getting Location ... ")
+        send_url = 'http://api.ipstack.com/check?access_key=f16ebe59174140a634827d674e605350&output=json&legacy=1'
```
Comment (pnhofmann) [inline jarviscli/plugins/name_day.py:101]: Don't really like api keys in source code...
I'll think of something - Jarvis dev branch has (or will have) support for getting location build directly into core.

I will come back to your PR later (and should I forget, comment something ;) )

Comment (AirlineDog) [reply jarviscli/plugins/name_day.py:101]: I took the code from here, but i genetated my own free API key
https://github.com/sukeesh/Jarvis/blob/2ed8cbd0655b4700da931962afd78747282b56dc/jarviscli/packages/mapps.py#L13-L15
The same API can be found here:
https://github.com/sukeesh/Jarvis/blob/700ea2e8b1f88aca5d5baf5df0103c9ec647fe15/jarviscli/plugins/location.py#L12

Notice that when i tried to run the two above code samples, the first one returned : `{'success': False, 'error': {'code': 104, 'type': 'usage_limit_reached', 'info': 'Your monthly usage limit has been reached. Please upgrade your Subscription Plan.'}}` and the second one worked fine.

**Comment by AirlineDog** [comment]
As a solution to your request i see two possible fixes:
1. Remove the `get_location()` functionality and instead ask the user for his location directly.
2. Divide the existing [location plugin](https://github.com/sukeesh/Jarvis/blob/700ea2e8b1f88aca5d5baf5df0103c9ec647fe15/jarviscli/plugins/location.py) into two methods. The first one will return a json object containing user's location info, that i will also call from my plugin, and the second one will do all the appropriate processing and printing.

About the [mapps](https://github.com/sukeesh/Jarvis/blob/2ed8cbd0655b4700da931962afd78747282b56dc/jarviscli/packages/mapps.py) package that returns an error, that is because the ipstack api only allows 100 requests per month for the [free plan](https://ipstack.com/product) (which probably is the case). This is also true for my api key

If you like any of my fixes or if you have any other suggestion please let me know to start implementing it.

**Comment by pnhofmann** [comment]
Sorry for the delay

> 2. Divide the existing [location plugin](https://github.com/sukeesh/Jarvis/blob/700ea2e8b1f88aca5d5baf5df0103c9ec647fe15/jarviscli/plugins/location.py)

That's an argument - the api key problem exists even without your PR.

Yeah, actually experimented on a location function directly implemented into Jarvis
https://github.com/pnhofmann/Jarvis/blob/dev/jarviscli/packages/geolocation.py

=> So, I'm deciding, wouldn't make much sense for you to change anything. I should build this feature probably. So, I just opened an issue for me: #971
And let's merge your PR ;)

**Comment by pnhofmann** [comment]
Sorry for the delay

> 2. Divide the existing [location plugin](https://github.com/sukeesh/Jarvis/blob/700ea2e8b1f88aca5d5baf5df0103c9ec647fe15/jarviscli/plugins/location.py)

That's an argument - the api key problem exists even without your PR.

Yeah, actually experimented on a location function directly implemented into Jarvis
https://github.com/pnhofmann/Jarvis/blob/dev/jarviscli/packages/geolocation.py

=> So, I'm deciding, wouldn't make much sense for you to change anything. I should build this feature probably. So, I just opened an issue for me: #971
And let's merge your PR ;)

**Pull request merged**